gate-saml/src/main/java/com/netflix/spinnaker/gate/security/saml/SecuritySamlProperties.java

```diff
@@ -13,88 +12,116 @@
-  private String redirectProtocol = "https";
-  private String redirectHostname;
-  private String redirectBasePath = "/";
-  // The application identifier given to the IdP for this app.
-  private String issuerId;
 
+  /** Registration id for this SAML provider. Used in SAML processing URLs. */
+  @NotEmpty private String registrationId = "SSO";
+
+  /**
+   * The Relying Party's entity ID (sometimes called an issuer ID). The value may contain a number
+   * of placeholders. They are "baseUrl", "registrationId", "baseScheme", "baseHost", and
+   * "basePort".
+   */
+  @NotEmpty private String issuerId = "{baseUrl}/saml2/metadata";
+
+  /**
+   * The path used for login processing. When combined with the base URL, this should form the
+   * assertion consumer service location.
+   */
+  @NotEmpty private String loginProcessingUrl = "/saml/{registrationId}";
+
+  /**
+   * Returns the assertion consumer service location template to use for redirecting back from the
+   * identity provider.
+   */
+  public String getAssertionConsumerServiceLocation() {
+    return "{baseUrl}" + loginProcessingUrl;
+  }
+
+  /** Optional list of roles required for authentication to succeed. */
   private List<String> requiredRoles;
+
+  /** Determines whether to sort the roles returned from the SAML provider. */
   private boolean sortRoles = false;
+
+  /** Toggles whether role names should be converted to lowercase. */
   private boolean forceLowercaseRoles = true;
 
-  @NestedConfigurationProperty
+  @Nonnull @NestedConfigurationProperty
   private UserAttributeMapping userAttributeMapping = new UserAttributeMapping();
 
-  private long maxAuthenticationAge = 7200;
-
-  // SHA1 is the default registered in DefaultSecurityConfigurationBootstrap.populateSignatureParams
-  private String signatureDigest = "SHA1";
-
-  public SignatureDigest signatureDigest() {
-    return SignatureDigest.fromName(signatureDigest);
-  }
-
-  /**
-   * Ensure that the keystore exists and can be accessed with the given keyStorePassword and
-   * keyStoreAliasName. Validates the configured signature/digest is supported.
-   */
   @PostConstruct
   public void validate() throws IOException, GeneralSecurityException {
     if (StringUtils.hasLength(metadataUrl) && metadataUrl.startsWith("/")) {
-      metadataUrl = FILE_SCHEME + metadataUrl;
+      metadataUrl = "file:" + metadataUrl;
     }
-    if (StringUtils.hasLength(keyStore)) {
-      if (!keyStore.startsWith(FILE_SCHEME)) {
-        keyStore = FILE_SCHEME + keyStore;
+    if (keyStore != null) {
+      if (keyStoreType == null) {
+        keyStoreType = "PKCS12";
       }
-      var path = Path.of(URI.create(keyStore));
       var keystore = KeyStore.getInstance(keyStoreType);
       var password = keyStorePassword != null ? keyStorePassword.toCharArray() : new char[0];
-      try (var stream = Files.newInputStream(path)) {
+      try (var stream = Files.newInputStream(keyStore)) {
```

Comment:
Looks like this is missing where the keyStore would by /some/pkcs.pkcs12 file change to file://some/pkcs.pkcs12?  AKA the FILE_SCHEME Block is removed and this is a change in that parsing.  Is that file prefix added someplace else or did that newInputStream handle it without the file:// scheme?